src/ecs.rs

```diff
@@ -0,0 +1,692 @@
+    /// It is the highest-level event categorization in ECS.
+    ///
+    /// # Allowed values
+    ///
+    /// - alert
+    /// - enrichment
+    /// - event
+    /// - metric
+    /// - state
+    /// - pipeline_error
+    /// - signal
+    #[serde(skip_serializing_if = "Option::is_none")]
+    kind: Option<String>,
+    /// Category represents a wider range of classification under ECS. Notably it is an array to
+    /// allow for events that may be categorized under multiple values.
+    ///
+    /// # Allowed values
+    ///
+    /// - api
+    /// - authentication
+    /// - configuration
+    /// - database
+    /// - driver
+    /// - email
+    /// - file
+    /// - host
+    /// - iam
+    /// - intrusion_detection
+    /// - library
+    /// - malware
+    /// - network
+    /// - package
+    /// - process
+    /// - registry
+    /// - session
+    /// - threat
+    /// - vulnerability
+    /// - web
+    #[serde(skip_serializing_if = "Option::is_none")]
+    category: Option<Vec<String>>,
+    /// Type represents a categorization "sub-bucket" that allows events to be differentiated withing a
+    /// category. Notably it is an array to allow for events that may be categorized under multiple values.
+    ///
+    /// # Allowed values
+    ///
+    /// - access
+    /// - admin
+    /// - allowed
+    /// - change
+    /// - connection
+    /// - creation
+    /// - deletion
+    /// - denied
+    /// - end
+    /// - error
+    /// - group
+    /// - indicator
+    /// - info
+    /// - installation
+    /// - protocol
+    /// - start
+    /// - user
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        rename(serialize = "type", deserialize = "type")
+    )]
+    type_: Option<Vec<String>>,
+    /// Outcome simply denotes whether the event represents a success or a failure from the perspective of
+    /// the entity that produced the event. It is the lowest-level event categorization in ECS.
+    ///
+    /// # Allowed values
+    ///
+    /// - failure
+    /// - success
+    /// - unknown
+    #[serde(skip_serializing_if = "Option::is_none")]
+    outcome: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsLogFieldSet {
+    /// Original log level of the log event.
+    level: String,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsHttpFieldSet {
+    /// Fields related to an HTTP request.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    request: Option<EcsHttpRequestFieldSet>,
+    /// Fields related to an HTTP response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    response: Option<EcsHttpResponseFieldSet>,
+    /// HTTP version.
+    ///
+    /// # Example
+    ///
+    /// `1.1`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    version: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug)]
+pub(crate) struct EcsHttpRequestFieldSet {
```

Comment:
That's correct, this is deliberate. It may be optionally supported in the future, as it's a very useful feature, but it's tricky to do it right without significant security and privacy impact. e.g., Oops, we logged everyone's passwords. Extremely reliable auto-redaction and things like only logging a body for requests crossing some detection threshold are both prerequisites. I'll probably implement body hashes before implementing full body logging. Additionally, there'd need to be upper maximums for how many bytes of the body to log.